src/Logic.hs

```diff
@@ -596,31 +583,23 @@ synchronizeState stateInitial =
 proceed :: ProjectState -> Action ProjectState
 proceed state = do
   state' <- provideFeedback state
-  case (Pr.getIntegrationCandidates state', Pr.candidatePullRequests state') of
+  case (Pr.integratedPullRequests state', Pr.candidatePullRequests state') of
                            -- Proceed with an already integrated candidate
     (candidate:_, _)    -> proceedCandidate candidate state'
                            -- Found a new candidate, try to integrate it.
     (_,           pr:_) -> tryIntegratePullRequest pr state'
                            -- No pull requests eligible, do nothing.
     (_,           _)    -> return state'
 
--- TODO: Get rid of the tuple; just pass the ID and do the lookup with fromJust.
-proceedCandidate :: (PullRequestId, PullRequest) -> ProjectState -> Action ProjectState
-proceedCandidate (pullRequestId, pullRequest) state =
-  case Pr.integrationStatus pullRequest of
-    NotIntegrated -> pure state -- dead code / unreachable
-
-    Integrated sha buildStatus -> case buildStatus of
-      BuildPending _ -> pure state
-      BuildSucceeded -> pushCandidate (pullRequestId, pullRequest) sha state
-                        -- If the build failed, give feedback on the PR
-      BuildFailed _  -> pure $ Pr.setNeedsFeedback pullRequestId True state
```

Comment:
Ah, so instead of always setting this to true when there's activity on a failed PR, we now only do this when the state initially changes?

Reply:
Yes.  Before, for a `BuildFailed` candidate:

1. set PR build to failed when event is received
2. go through the event loop
3. in `proceedCandidate`, "discover" and set that the PR needs feedback
4. go through the event loop again
5. post comment (for the PR that needs feedback)

Now:

1. set PR build to failed and that it needs feedback
2. go through the event loop
3. post comment (for the PR that needs feedback)